bowjoin.go

```diff
@@ -375,8 +389,30 @@ func outerFillLeftBowCols(newSeries *[]Series, left, right *bow, newNumRows, uni
 					break
 				}
 			}
+		case TimestampSec, TimestampMilli, TimestampMicro, TimestampNano:
+			data := array.NewTimestampData(left.Column(colIndex).Data())
```

Comment:
Could we extract this code block on function(s) ?

Reply:
Maybe you can create functions for sub-blocks and not the entire `case` ?

Reply:
This is not easy to split, as we have two nested for loops with many variables. The logic here is already quite complex...

Reply:
This is also out of the scope of this PR

Reply:
Next time don't hesitate to write tests first and refactor earlier, it often avoids a kind of code complexity